scalyr_agent/scalyr_client.py

```diff
@@ -1543,39 +1541,49 @@ def __set_attributes(self, thread_id, attributes):
         self.__serialization_base = tmp_buffer.getvalue()
 
     def __get_attributes_to_serialize(self):
+        """Return the attributes that should be included in the serialization for this specific event.
+        This does not include attributes from its parent if using the logfile_addevents_format, and does not include
+        attributes from this event that already exist in the parent with the same value.
+        """
+        result = dict()
         if self.__disable_logfile_addevents_format:
-            result = dict()
             if self.__parent_event:
                 result = dict(self.__parent_event.__attrs)
             if self.__attrs:
                 result.update(self.__attrs)
-            return result
         else:
-            return self.__attrs
+            if self.__parent_event:
+                for key in self.__attrs:
+                    if (
+                        key not in self.__parent_event.__attrs
+                        or self.__parent_event.__attrs[key] != self.__attrs[key]
+                    ):
+                        result[key] = self.__attrs[key]
+            else:
+                result = self.__attrs
+        return result
 
     def add_attributes(self, attributes, overwrite_existing=False):
         """ Adds items attributes to __attrs if the __parent_event doesn't
         already have those attributes set.
 
-        If overwrite_existing is True an attribute will be added even if the __parent_event has the same key defined,
-        unless the value also matched to avoid wasting bytes.
+        If overwrite_existing is False an attribute will not be added if the key already exists in __attrs.
         """
         if attributes:
             attributes = dict(attributes)
 
         if self.__parent_event:
```

Comment:
Remove this.. no need to check `__parent_event`.  This method should operate the same way either way.